android_ndk/jni/Android.mk

```diff
@@ -0,0 +1,65 @@
+#
+# vkrunner
+#
+# Copyright (C) 2018 Google LLC
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice (including the next
+# paragraph) shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+#
+
+LOCAL_PATH := $(call my-dir)
+
+ROOT_PATH = $(LOCAL_PATH)/../..
+VKRUNNER_PATH = $(ROOT_PATH)/vkrunner
+
+include $(CLEAR_VARS)
+
+LOCAL_MODULE    := vkrunner
+LOCAL_SRC_FILES := $(VKRUNNER_PATH)/vr-allocate-store.c \
+	$(VKRUNNER_PATH)/vr-box.c \
+	$(VKRUNNER_PATH)/vr-buffer.c \
+	$(VKRUNNER_PATH)/vr-config.c \
+	$(VKRUNNER_PATH)/vr-context.c \
+	$(VKRUNNER_PATH)/vr-error-message.c \
+	$(VKRUNNER_PATH)/vr-executor.c \
+	$(VKRUNNER_PATH)/vr-feature-offsets.c \
+	$(VKRUNNER_PATH)/vr-flush-memory.c \
+	$(VKRUNNER_PATH)/vr-format.c \
+	$(VKRUNNER_PATH)/vr-half-float.c \
+	$(VKRUNNER_PATH)/vr-hex.c \
+	$(VKRUNNER_PATH)/vr-list.c \
+	$(VKRUNNER_PATH)/vr-pipeline.c \
+	$(VKRUNNER_PATH)/vr-pipeline-key.c \
+	$(VKRUNNER_PATH)/vr-result.c \
+	$(VKRUNNER_PATH)/vr-script.c \
+	$(VKRUNNER_PATH)/vr-subprocess.c \
+	$(VKRUNNER_PATH)/vr-temp-file.c \
+	$(VKRUNNER_PATH)/vr-test.c \
+	$(VKRUNNER_PATH)/vr-util.c \
+	$(VKRUNNER_PATH)/vr-vbo.c \
+	$(VKRUNNER_PATH)/vr-vk.c \
+	$(VKRUNNER_PATH)/vr-window.c
+
```

Comment:
It looks like the Android NDK has support for building with CMake. Would it not be possible to use this to generate the build files from the CMake files so that we don’t have to maintain two lists of source files?

https://developer.android.com/ndk/guides/cmake

Reply:
That CMake support is in the Gradle build system in Android Studio, and is relatively new.  The NDK is a lower level, and the Android.mk build system will work with all Android apps whether they are built with Gradle or only with the NDK.
 (Android.mk is GNU Make with extra macros and functions)

Reply:
Both CMake and Android.mk require Android NDK. Thus, building it using Android.mk does not give more required programs and it is more popular for Android build.